warehouse/query-core/src/main/java/datawave/query/index/lookup/CreateUidsIterator.java

```diff
@@ -104,27 +143,40 @@ public boolean hasTop() {
     public void next() throws IOException {
         tk = null;
         if (src.hasTop()) {
+            
+            // Strip the datatype from the CQ, leaving just the shard
             Key reference = makeRootKey(src.getTopKey());
+            
             List<String> uids = Lists.newLinkedList();
             long count = 0L;
-            boolean ignore = false;
-            if (collapseUids) {
-                ignore = true;
-            }
+            boolean ignore = collapseUids;
+            
             while (src.hasTop() && sameShard(reference, src.getTopKey())) {
-                Key nextTop = src.getTopKey();
-                Tuple3<Long,Boolean,List<String>> uidInfo = parseUids(nextTop, src.getTopValue());
+                
+                Key key = src.getTopKey();
+                Value value = src.getTopValue();
+                
+                Tuple3<Long,Boolean,List<String>> uidInfo = parseUids(key, value);
                 count += uidInfo.first();
                 ignore |= uidInfo.second();
-                if (!ignore)
-                    for (String uid : uidInfo.third()) {
-                        if (log.isTraceEnabled())
-                            log.trace("Adding uid " + StringUtils.split(uid, '\u0000')[1]);
-                        uids.add(uid);
-                    }
+                
+                // Three things must be true in order to return a document specific range
+                // 1. The uids stored in the index must have the ignore flag set to FALSE
+                // 2. The option to collapse all document ranges must be false.
+                // 3. The document count must be under the document threshold.
+                if (ignore || !collapseUids || (count <= collapseUidsThreshold)) {
```

Comment:
this used to be if (!ignore).  Are we sure we got this right?

Reply:
Good catch, that should be !ignore.